src/test/kotlin/fr/social/gouv/agora/infrastructure/feedbackQag/repository/GetFeedbackQagRepositoryImplTest.kt

```diff
@@ -0,0 +1,125 @@
+import java.util.*
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest
+internal class GetFeedbackQagRepositoryImplTest {
+
+    @Autowired
+    private lateinit var repository: GetFeedbackQagRepositoryImpl
+
+    @MockBean
+    private lateinit var databaseRepository: FeedbackQagDatabaseRepository
+
+    @MockBean
+    private lateinit var cacheRepository: FeedbackQagCacheRepository
+
+    private val feedbackQagDTO = FeedbackQagDTO(
+        id = UUID.randomUUID(),
+        userId = "userId",
+        qagId = UUID.randomUUID(),
+        isHelpful = false,
+    )
+
+    @Test
+    fun `getFeedbackQag - when invalid qagId - should return null`() {
+        // When
+        val result = repository.getFeedbackQagStatus(
+            FeedbackQag(
+                qagId = "invalid qagId UUID",
+                userId = "userId",
+            )
+        )
+
+        // Then
+        assertThat(result).isNull()
+        then(cacheRepository).shouldHaveNoInteractions()
+        then(databaseRepository).shouldHaveNoInteractions()
+    }
+
+    @Test
+    fun `getFeedbackQag - when CacheNotInitialized - should call getFeedbackQag from database and insert result to cache`() {
+        // Given
+        val qagUUID = UUID.randomUUID()
+        given(cacheRepository.getFeedbackQag(qagId = qagUUID, userId = "userId"))
+            .willReturn(CacheResult.CacheNotInitialized)
+
+        given(databaseRepository.getFeedbackQag(qagId = qagUUID, userId = "userId")).willReturn(null)
+
+        // When
+        val result = repository.getFeedbackQagStatus(
+            FeedbackQag(
+                qagId = qagUUID.toString(),
+                userId = "userId",
+            )
+        )
+
+        // Then
+        assertThat(result?.isExist).isEqualTo(false)
+        inOrder(cacheRepository).also {
+            then(cacheRepository).should(it).getFeedbackQag(qagId = qagUUID, userId = "userId")
+            then(cacheRepository).should(it)
+                .insertFeedbackQag(qagId = qagUUID, userId = "userId", feedbackQagDTO = null)
+            it.verifyNoMoreInteractions()
+        }
+        then(databaseRepository).should(times(1)).getFeedbackQag(qagId = qagUUID, userId = "userId")
+    }
+
+    @Test
+    fun `getFeedbackQag - when CachedFeedbackQagNotFound - should return object from cache`() {
+        // Given
+        val qagUUID = UUID.randomUUID()
+        given(cacheRepository.getFeedbackQag(qagId = qagUUID, userId = "userId"))
+            .willReturn(CacheResult.CachedFeedbackQagNotFound)
+
+        // When
+        val result = repository.getFeedbackQagStatus(
+            FeedbackQag(
+                qagId = qagUUID.toString(),
+                userId = "userId",
+            )
+        )
+
+        // Then
+        assertThat(result?.isExist).isEqualTo(false)
+        then(cacheRepository).should(times(1)).getFeedbackQag(qagId = qagUUID, userId = "userId")
+        then(cacheRepository).shouldHaveNoMoreInteractions()
```

Comment:
Ces 2 lignes se remplacent par:
`then(cacheRepository).should(only()).getFeedbackQag(qagId = qagUUID, userId = "userId")`

Reply:
done